src/main/java/org/kiwiproject/eureka/EmbeddedEurekaBootstrap.java

```diff
@@ -1,15 +1,116 @@
 package org.kiwiproject.eureka;
 
+import static java.util.Objects.nonNull;
+
+import com.netflix.appinfo.InstanceInfo;
 import com.netflix.discovery.DiscoveryClient;
+import com.netflix.discovery.shared.Application;
 import com.netflix.eureka.Jersey2EurekaBootStrap;
+import com.netflix.eureka.registry.AbstractInstanceRegistry;
+import com.netflix.eureka.util.MeasuredRate;
+import lombok.extern.slf4j.Slf4j;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
 
-// TODO: Add javadoc
-// TODO: Can this be package private?  People will get one of these in tests, but don't need to create one.
+/**
+ * Extension of Eureka's bootstrap but with added accessors to validate internal Eureka data.
+ */
+@Slf4j
 public class EmbeddedEurekaBootstrap extends Jersey2EurekaBootStrap {
 
     public EmbeddedEurekaBootstrap(DiscoveryClient client) {
         super(client);
     }
 
-    // TODO: Add various calls to access/modify internal Eureka data :-)
+    /**
+     * Cleans out all the registered applications inside of Eureka.
+     */
+    public void cleanupApps() {
+        LOG.info("Clearing registry");
+        serverContext.getRegistry().clearRegistry();
+    }
+
+    /**
+     * Returns a list of all the registered applications in Eureka.
+     *
+     * @return list of registered applications.
+     */
+    public List<Application> registeredApplications() {
+        return serverContext.getRegistry().getApplications().getRegisteredApplications();
+    }
+
+    /**
+     * Checks to see if a given app is registered in Eureka.
+     *
+     * @param appId the id of the app to check.
+     * @return true if registered, false otherwise.
+     */
+    public boolean isApplicationRegistered(String appId) {
+        return nonNull(serverContext.getRegistry().getApplication(appId));
+    }
+
+    /**
+     * Returns an {@link Application} with given appId if found from Eureka.
+     *
+     * @param appId the id of the app to check.
+     * @return the registered application if found, otherwise null.
+     */
+    public Application getRegisteredApplication(String appId) {
+        return serverContext.getRegistry().getApplication(appId);
+    }
+
+    /**
+     * Returns the count of heartbeat renewals in Eureka.
+     *
+     * @return the number of heartbeats sent to Eureka
+     * @implNote This requires reflection hacks due to the way Eureka stores the renews and what is visible from
+     * this class.  I don't like it, but it is what it is.
+     */
+    @SuppressWarnings("java:S3011")
+    public long getHeartbeatCount() {
+        try {
+            var renewsInLastMinField = AbstractInstanceRegistry.class.getDeclaredField("renewsLastMin");
```

Comment:
I must be missing something...why not just use:

```java
serverContext.getRegistry().getNumOfRenewsInLastMin()
```

Reply:
Because it goes into `MeasureRate` and pulls from the lastBucket, but when the heartbeat goes in, it increments the currentBucket, then there is a job that runs every minute to move the currentBucket to the lastBucket 😢

Reply:
```java
    /**
     * Returns the count in the last sample interval.
     */
    public long getCount() {
        return lastBucket.get();
    }

    public synchronized void start() {
        if (!isActive) {
            timer.schedule(new TimerTask() {

                @Override
                public void run() {
                    try {
                        // Zero out the current bucket.
                        lastBucket.set(currentBucket.getAndSet(0));
                    } catch (Throwable e) {
                        logger.error("Cannot reset the Measured Rate", e);
                    }
                }
            }, sampleInterval, sampleInterval);

            isActive = true;
        }
    }
```

Reply:
wow ok, then I think your explanation above (without the code) should be included...

Reply:
...otherwise someone like me will ask the same question 😂